prime-angular-frontend/src/app/modules/adjudication/shared/components/site-registration-container/site-registration-container.component.ts

```diff
@@ -0,0 +1,161 @@
+
+@Component({
+  selector: 'app-site-registration-container',
+  templateUrl: './site-registration-container.component.html',
+  styleUrls: ['./site-registration-container.component.scss']
+})
+export class SiteRegistrationContainerComponent implements OnInit {
+  @Input() public hasActions: boolean;
+  @Input() public content: TemplateRef<any>;
+  @Output() public action: EventEmitter<void>;
+
+  public busy: Subscription;
+  public columns: string[];
+  public dataSource: MatTableDataSource<Site>;
+
+  public showSearchFilter: boolean;
+  public AdjudicationRoutes = AdjudicationRoutes;
+
+  private routeUtils: RouteUtils;
+
+  constructor(
+    @Inject(DIALOG_DEFAULT_OPTION) private defaultOptions: DialogDefaultOptions,
+    private route: ActivatedRoute,
+    private router: Router,
+    private authService: AuthService,
+    private organizationResource: OrganizationResource,
+    private siteResource: SiteResource,
+    private dialog: MatDialog
+  ) {
+    this.routeUtils = new RouteUtils(route, router, AdjudicationRoutes.MODULE_PATH);
+
+    this.action = new EventEmitter<void>();
+
+    this.hasActions = false;
+    this.dataSource = new MatTableDataSource<Site>([]);
+  }
+
+  public onSearch(search: string | null): void {
+    this.routeUtils.updateQueryParams({ search });
+  }
+
+  public onFilter(status: any | null): void {
+    this.routeUtils.updateQueryParams({ status });
+  }
+
+  public onRefresh(): void {
+    // Use existing query params for initial search
+    this.getDataset(this.route.snapshot.queryParams);
+
+    // Update results on query param change
+    this.route.queryParams
+      .subscribe((queryParams: { [key: string]: any }) => this.getDataset(queryParams));
+  }
+
+  public onViewOrganization(siteId: number) {
+    // TODO look up organization
+    this.routeUtils.routeRelativeTo([]);
+  }
+
+  public onViewSite(siteId: number) {
+    this.routeUtils.routeRelativeTo([siteId]);
+  }
+
+  public onDelete(siteId: number) {
+    const data = this.defaultOptions.delete('site');
+    if (this.authService.isSuperAdmin()) {
+      this.busy = this.dialog.open(ConfirmDialogComponent, { data })
+        .afterClosed()
+        .pipe(
+          exhaustMap((result: boolean) =>
+            (result)
+              ? of(noop)
+              : EMPTY
+          ),
+          exhaustMap(() => this.siteResource.deleteSite(siteId)),
+          map((site: Site) => this.dataSource.data = MatTableDataSourceUtils.delete<Site>(this.dataSource, 'id', site.id))
+        )
+        .subscribe(() => this.routeUtils.routeRelativeTo(['./']));
+    }
+  }
+
+  public onRoute(routePath: string | (string | number)[]) {
+    this.routeUtils.routeWithin(routePath);
+  }
+
+  public ngOnInit(): void {
+    this.getDataset(this.route.snapshot.queryParams);
+  }
+
+  // private getDataset(queryParams: { search?: string, status?: number }): void {
```

Comment:
Note for reviewers.  Changes in requirements dropped using organizations so commented out, and kept until fruitloops next sprint